### PR TITLE
Bug 1267553 - Unify system and search places database r=benfrancis

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9d5a3fbb83b06a1607e5d91b79394eda8b95246

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6288cfee8c57ff3e1f0874fe3e1bd0eb15671488

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a93d470afa395041678e7ce83a74116f5f675c0f

File: apps/homescreen/js/apps.js [hunk 1]
```diff
@@ -165,7 +165,7 @@
     init: function() {
       this.icons.classList.toggle('small', this.settings.small);
       this.scrollable.classList.toggle('snapping',
-                                       this.settings.scrollSnapping);
+        this.settings.scrollSnapping);
```
Comment (albertopq) [inline apps/homescreen/js/apps.js]: This is for fixing JSHINT

File: apps/system/js/places.js [hunk 2]
```diff
@@ -33,10 +33,11 @@
      * @memberof Places.prototype
      * @type {String}
      */
-    DB_NAME: 'places',
-    DB_VERSION: 1,
+    DB_NAME: 'places_idb_store',
```
Comment (benfrancis) [inline apps/system/js/places.js]: Can we keep 'places'?

File: apps/system/js/places.js [hunk 3]
```diff
@@ -137,36 +138,36 @@
...
-      if(!this.db.objectStoreNames.contains(this.SITES_STORE)) {
-        var sitesStore = this.db.createObjectStore(this.SITES_STORE,
-          { keyPath: 'id', autoIncrement: false});
+
+    upgrade: function(e) {
+      this.db = e.target.result;
+      var fromVersion = e.oldVersion;
+
+      if (fromVersion < 1) {
+        var places = this.db.createObjectStore(this.PAGES_STORE, {
```
Comment (benfrancis) [inline apps/system/js/places.js]: "pages" rather than "places"?

File: apps/system/js/places.js [hunk 4]
```diff
@@ -137,36 +138,36 @@
...
+
+      if (fromVersion < 2) {
+        asyncStorage.removeItem('latest-revision');
+        var visits = this.db.createObjectStore(this.VISITS_STORE, {
+          keyPath: 'date'
+        });
+        visits.createIndex('date', 'date', { unique: true });
       }
 
+      if (fromVersion < 3) {
```
Comment (benfrancis) [inline apps/system/js/places.js]: Can this be combined with version 2?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a78b6dfd449cbeee2effb573ad35dddbde2f6676

**Pull request merged**